straxen/analyses/waveform_plot.py

```diff
@@ -338,3 +344,271 @@ def plot_wf(st: strax.Context,
+                           cmap=color_map)
+        # Is there a better way to do this update?
+        hp_opt.update(_common_opt)
+        _update = hp_opt.copy()
+        hp_opt.update(_update)
+
+    # S1
+    if events['s1_area'] > 0:
+        plt.sca(ax_s1)
+        context.plot_peaks(run_id,
+                           time_range=(events['s1_time'] - s1_fuzz,
+                                       events['s1_endtime'] + s1_fuzz),
+                           single_figure=False)
+
+        # Hit pattern plots
+        area = context.get_array(run_id, 'peaklets',
+                                 time_range=(events['s1_time'],
+                                             events['s1_endtime']),
+                                 keep_columns=('area_per_channel', 'time', 'dt', 'length'))
+        for ax, array in ((ax_s1_hp_t, 'top'), (ax_s1_hp_b, 'bottom')):
+            plt.sca(ax)
+            straxen.plot_on_single_pmt_array(c=np.sum(area['area_per_channel'], axis=0),
+                                             array_name=array,
+                                             **s1_hp_kwargs)
+            # Mark reconstructed position
+            plt.scatter(event['x'], event['y'], marker='X', s=100, c='r')
+
+    # S2
+    if events['s2_area'] > 0:
+        plt.sca(ax_s2)
+        context.plot_peaks(run_id,
+                           time_range=(events['s2_time'] - s2_fuzz,
+                                       events['s2_endtime'] + s2_fuzz),
+                           single_figure=False)
+
+        # Hit pattern plots
+        area = context.get_array(run_id, 'peaklets',
+                                 time_range=(events['s2_time'],
+                                             events['s2_endtime']),
+                                 keep_columns=('area_per_channel', 'time', 'dt', 'length'))
+        for ax, array in ((ax_s2_hp_t, 'top'), (ax_s2_hp_b, 'bottom')):
+            plt.sca(ax)
+            straxen.plot_on_single_pmt_array(c=np.sum(area['area_per_channel'], axis=0),
+                                             array_name=array,
+                                             **s2_hp_kwargs)
+            # Mark reconstructed position
+            plt.scatter(event['x'], event['y'], marker='X', s=100, c='r')
+
+    # Fill panels with peak/event info
+    for it, (ax, labels_and_unit) in enumerate([(ax_event_info, display_event_info),
+                                                (ax_peak_info, display_peak_info)]):
+        for i, (_lab, _unit) in enumerate(labels_and_unit):
+            coord = 0.01, 0.9 - 0.9 * i / len(labels_and_unit)
+            ax.text(*coord, _lab[:24], va='top', zorder=-10)
+            ax.text(coord[0] + 0.5, coord[1],
+                    _unit.format(v=event[_lab]), va='top', zorder=-10)
+            # Remove axes and labels from panel
+            ax.set_xticks([])
+            ax.set_yticks([])
+            _ = [s.set_visible(False) for s in ax.spines.values()]
```

Comment:
I hope you do not have to add additional parameters, this looks quite static, but I could be wrong.

Reply:
Actually, it is flexible (till some level), you just need to change the `display_event_info`  argument as in the docstring. It will just divides the panel in whatever length info you need and evenly spaces it there. Could do something smarter if needed but I guess this is fine for now.

Reply:
If you want to see how that works, you can look at the difference between two of the examples I've posted, the event-info panel has been slightly updated with more info, and the lines of attributes are scaled accordingly.